components/ai_chat/resources/page/components/input_box/index.tsx

```diff
@@ -45,84 +52,101 @@ function InputBox(props: Props) {
   const handleOnKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
     if (e.key === 'Enter' && !e.shiftKey && !e.nativeEvent.isComposing) {
       if (!e.repeat) {
-        props.submitInputTextToAPI()
+        props.context.submitInputTextToAPI()
       }
 
       e.preventDefault()
     }
 
     if (
       e.key === 'Backspace' &&
-      props.inputText === '' &&
-      props.selectedActionType
+      props.context.inputText === '' &&
+      props.context.selectedActionType
     ) {
-      props.resetSelectedActionType()
+      props.context.resetSelectedActionType()
+    }
+  }
+
+  const handleFocus = (event: React.FormEvent<HTMLTextAreaElement>) => {
+    if (!props.context.isMobile) {
```

Comment:
I think it would be better to always fire these events (including on desktop) and let the `isMobile` filter be part of the event handler which is passed in, if it's important.

This way splits the logic in two different places, which might be a bit unexpected as normally a focus/blur handler will be passed through transparently.

Reply:
that would turn into unneeded calls on desktop, we don't need them there currently

Reply:
I think the handlers at the top level should make that decision. Alternatively, we could rename the event in props to `onFocusMobile` to make it clear the event will only fire on mobile.

If this is for performance reasons, then maybe it would be better to not attach the event handlers at all (perhaps setting them to null/undefined)?